src/backend/utils/adt/numeric.c

```diff
@@ -5995,6 +6004,14 @@ numeric_sum(PG_FUNCTION_ARGS)
 
 	init_var(&sumX_var);
 	accum_sum_final(&state->sumX, &sumX_var);
+
+	if (sql_dialect == SQL_DIALECT_TSQL &&
+	    (*exact_numeric_digit_count_hook)(sumX_var.weight, sumX_var.dscale, sumX_var.digits[0], DEC_DIGITS) >
```

Comment:
we should abstract the overflow detection inside hook, say detect_numeric_overflow which returns true or false. That way TSQLMaxNumPrecision is not used on engine side

Reply:
done